pkg/storagepool/cstorpool.go

```diff
@@ -0,0 +1,161 @@
+type cstorPoolOperationOptions struct {
+	// runNamespace is the namespace where cstorPool operation will happen
+	//runNamespace string
+	// k8sClient will make K8s API calls
+	k8sClient *m_k8s_client.K8sClient
+}
+
+// cstorPoolOperation exposes methods with respect to cstorPool related operations
+// e.g. read, create, delete.
+type cstorPoolOperation struct {
+	// cstorPoolOperationOptions has the options to various cstorPool related
+	// operations
+	cstorPoolOperationOptions
+	// cstorPool to create or read or delete
+	cstorPool *v1alpha1.CStorPool
+}
+
+// NewCstorPoolOperation returns a new instance of cstorPoolOperation
+func NewCstorPoolOperation(cstorPool *v1alpha1.CStorPool) (*cstorPoolOperation, error) {
+	if cstorPool == nil {
+		return nil, fmt.Errorf("failed to instantiate cstorPool operation: nil cstorPool was provided")
+	}
+
+	kc, err := m_k8s_client.NewK8sClient(cstorPool.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	// Put cstor pool object inside cstorPoolOperation object
+	return &cstorPoolOperation{
+		cstorPool: cstorPool,
+		cstorPoolOperationOptions: cstorPoolOperationOptions{
+			k8sClient: kc,
+		},
+	}, nil
+}
+
+// Create provisions an OpenEBS cstorPool
+func (v *cstorPoolOperation) Create() (*v1alpha1.CStorPool, error) {
+	if v.k8sClient == nil {
+		return nil, fmt.Errorf("unable to create cstorPool: nil k8s client")
+	}
+
+	castName := v.cstorPool.Annotations[string(v1alpha1.SPCASTemplateCK)]
+	if len(castName) == 0 {
+		//return nil, fmt.Errorf("unable to create cstorPool: missing create cas template at '%s'", v1alpha1.CASTemplateCVK)
+		return nil, fmt.Errorf("unable to create cstorPool: missing create cas template")
+	}
+
+	// fetch CASTemplate specifications
+	//cast, err := v.k8sClient.GetOEV1alpha1CAST(castName, mach_apis_meta_v1.GetOptions{})
+	cast, err := v.k8sClient.GetOEV1alpha1CAST(castName, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// provision cas cstorPool via cas template engine
+	cc, err := NewCASStoragePoolEngine(
+		// ToDo : pvc and sc config should be striped off in context to pool
+		"null",
+		"null",
+		cast,
+		string(v1alpha1.CstorPoolTLP),
+		map[string]interface{}{
+			string(v1alpha1.OwnerCTP):    v.cstorPool.Name,
+			string(v1alpha1.PoolTypeCTP):     v.cstorPool.Spec.PoolSpec.PoolType,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the cstorPool
+	data, err := cc.Create()
+	if err != nil {
+		return nil, err
+	}
+
+	// unmarshall into openebs cstorPool
+	cstorPool := &v1alpha1.CStorPool{}
+	err = yaml.Unmarshal(data, cstorPool)
+	if err != nil {
+		return nil, err
+	}
+	return cstorPool, nil
+}
+
+func (v *cstorPoolOperation) Delete() (*v1alpha1.CStorPool, error) {
+	if len(v.cstorPool.Name) == 0 {
+		return nil, fmt.Errorf("unable to delete cstor pool: cstor pool name not provided")
+	}
+
+	// cas template to delete a cstor pool
+	// Need to pass it as env variable
+	castName := "cstor-pool-delete-cast"
```

Comment:
please create a todo/backlog to fetch the name of `cas template for delete` from SPC object.
So `pool name -> pool details -> spc details -> cas template name for delete`